client/me/purchases/manage-purchase/index.jsx

```diff
@@ -329,22 +331,17 @@ class ManagePurchase extends Component {
 			}
 		}
 
-		if ( this.shouldShowNonPrimaryDomainWarning() ) {
-			const onClick = event => {
+		const onClick = event => {
+			trackNavItemClick( text );
+
+			if ( this.shouldShowNonPrimaryDomainWarning() ) {
```

Comment:
What is `this` going to resolve to inside this function...? It may be simpler to either push `this` into a variable available to the closure, or push the items we're referencing on `this` to separate variables accessible to the closure.

(Please correct me if `this` has better semantics these days!)

Reply:
So, in arrow functions, `this` resolves to whatever `this` resolves to in the surrounding scope. Meaning, if `this` works outside the arrow function, it will work inside as well.

Reply:
TIL! And happily so -- `this` has always been a nightmare in JS, so making it be predictable with arrow functions is a *good thing*.